config/sample-idex.cfg

```diff
@@ -94,3 +100,70 @@ gcode:
     ACTIVATE_EXTRUDER EXTRUDER=extruder1
     SET_DUAL_CARRIAGE CARRIAGE=1
     SET_GCODE_OFFSET Y=15
+
+[gcode_macro calibrate_separation]
+# YOU MUST CALIBRATE YOUR ROTATION_DISTANCE BEFORE THIS IS USEFUL!!!
+# Configure this to draw 2 line segments perpendicular to the dual_carraige axis
+# that meet in the middle of the bed. Both lines should have the same coordinate
+# on the axis of the dual_carraige. If they don't print inline, adjust your endstop
+# settings.
+gcode:
+    G28
+    G90
+    M83
+    T0 ; test T0
+    G1 X120 Y150 Z.2 F4800
+    G1 Y75 E10
+    T1 ; test T1
+    G1 X120 Y0 Z.2
+    G1 Y75 E10
```

Comment:
A modified macro to test for alignment on both X & Y axis, as mentioned by @eddietheengineer 

You'll probably have to have a fine-tuning print in step 3 as well.

```suggestion
#######################################################################
# Tool alignment Calibration
#######################################################################

[gcode_macro calibrate_separation]
# YOU MUST CALIBRATE YOUR ROTATION_DISTANCE BEFORE THIS IS USEFUL!!!
# Configure this to draw 2 line segments perpendicular to the dual_carraige axis
# and 2 line segments parallel. Both sets of lines should have the same coordinate
# on the axis of the dual_carraige. If they don't print inline, adjust your endstop
# settings, or specify an offset on a carriage to line them up.
gcode:
    G28
    G90
    M83

    # X axis alignment
    T0 ; test T0
    G1 X120 Y140 Z.2 F4800
    G1 Y80 E10
    T1 ; test T1
    G1 X120 Y20 Z.2
    G1 Y80 E10

    # Y axis alignment
    T0 ; test T0
    G1 X140 Y160 Z.2 F4800
    G1 X80 E10
    T1 ; test T1
    G1 X20 Y160 Z.2
    G1 Y80 E10
```

Reply:
I'm not sure what the advantage of this macro is over the one I wrote. I was also planning on writing a macro that takes the axis limits of the carriages and divides them by 2 automatically so the user doesn't have to setup the macro. 

I also need to put in a thing about calibrating the Z axis. It won't be part of this macro, but it's something that'll need to happen

Reply:
All this was is to try and see if your hotends are vastly out in Y offset. Mine is (probably due to plastic print warping, or assembly variance) out about 0.12mm. But this seems to be very static.
The issue that had me hesitating on this is that the same-axis variations are very small, so we need maybe only a step3-equivalent for it.
You can probably ignore this proposal.

Calibrating Z, I basically did a paper test separately. The issue is my hotends are about 0.5mm different in Z which is crazy high, and the eva2 hotend system doesn't have an easy way of adjusting this.
I would like to get them within 0.05mm so that simultaneous printing is possible, and need a way of easily adjusting them. but that's my design flaw.
(but bed uneven-ness will in any case require a fat 0.3mm first layer to even have a chance of success in any case)

Reply:
I tried to make a generic macro for this that reads the axis from the config, but then realised that `printer.dual_carriage` doesn't exist for cartesion IDEX, only hybrid_core* IDEX :sob: 
And #4464 doesn't address it either. Meaning it doesn't add that functionality to cartesian IDEX printers either.